### PR TITLE
Introduce models for "GeoStyler Data"

### PR DESCRIPTION
**Pull request by chrismayer**
This adds a first version for the models of the "GeoStyler Data" package.

Root object ``Data`` is an aggregation of the schema of the data (comparable to DescribeFeatureType) based on [JSON Schema](http://json-schema.org/) and the data itself (example features) as [GeoJSON](http://geojson.org/).

### CODE REVIEW

**Comment by marcjansen** [comment]
Thanks!

**Comment by marcjansen** [comment]
Is this still `Work in progress`?

**Comment by chrismayer** [comment]
I expect some modifications since this can be seen as a draft for the models of the "GeoStyler Data" package. So I would appreciate feedback, which leads me to give this free for reviews.

**Comment by chrismayer** [comment]
Thanks for your feedback @KaiVolland. I just addressed your remarks:
  - Using more ``type`` definitions instead of ``class``
  - Included a GeoJSON lib

Additionally:
  - Move the sources to the new folder ``Model``

Would you please have another look?

File: package.json [hunk 1]
```diff
@@ -33,6 +33,7 @@
   },
   "devDependencies": {
     "ts-jest": "22.4.3",
+    "@types/geojson": "^7946.0.2",
```
Comment (KaiVolland) [inline package.json]: Please remove the `^`.

File: src/Type/DataSchema.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+ * @type DataSchema
+ */
+type DataSchema = {
+
+  /**
+   * Optional title for the described entity
+   *
+   * @type {string}
+   */
+  title: string | '';
```
Comment (KaiVolland) [inline src/Type/DataSchema.ts]: I think it should be `title?: string;` if its optional

File: src/Type/DataSchema.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+  /**
+   * Optional type definition for the described entity
+   *
+   * @type {string}
+   */
+  type: string | 'object';
```
Comment (KaiVolland) [inline src/Type/DataSchema.ts]: I think it should be `type?: string;` if its optional. `'object'` allready is a string.

File: src/Type/DataSchema.ts (hunk 2 above)
Comment (chrismayer) [reply src/Type/DataSchema.ts]: By this notation I wanted to ensure that ``title`` is always a String (= default value) and not ``null`` or ``undefined``.  Does this make sense?

File: src/Type/DataSchema.ts (hunk 3 above)
Comment (chrismayer) [reply src/Type/DataSchema.ts]: By this notation I wanted to ensure that ``type`` is always a String (= default value) and not ``null`` or ``undefined``.  Does this make sense?

File: src/Type/DataSchema.ts (hunk 2 above)
Comment (KaiVolland) [reply src/Type/DataSchema.ts]: I don't think it's possible to define default values for properties in types. I think your syntax would something like: title can be any string or the string ''...

File: src/Type/DataSchema.ts (hunk 3 above)
Comment (KaiVolland) [reply src/Type/DataSchema.ts]: see above

File: src/Type/DataSchema.ts (hunk 2 above)
Comment (KaiVolland) [reply src/Type/DataSchema.ts]: IMHO we should leave it as required for know. We can be more precise later on. So you may just remove the empty string thing.

Comment (chrismayer) [reply src/Type/DataSchema.ts]: Agree...

**Pull request merged**